### PR TITLE
Support values for which empty is a valid state in the source generator

### PR DESCRIPTION
**Pull request by rabuckley**
Currently applied to UriPath and UriAsciiPath.

It's unfortunate that we can't remove the `_initialized` bool because its on the interface, so that stays there for now.

### CODE REVIEW

File: gen/DSE.Open.Values.Generators/Model/ValueTypeSpec.cs [hunk 1]
```diff
@@ -21,6 +21,8 @@ internal abstract class ValueTypeSpec
 
     public string ContainedValueTypeName { get; set; } = default!;
 
+    public bool EmitEnsureIntialised { get; internal set; }
```
Comment (frankbuckley) [inline gen/DSE.Open.Values.Generators/Model/ValueTypeSpec.cs:24]: Should default to true?

File: gen/DSE.Open.Values.Generators/ValueTypesGenerator.cs [hunk 2]
```diff
@@ -186,17 +187,21 @@ or TypeNames.ComparableValueAttributeFullName
...
                     }
 
-                    var maxSerializedCharLengthSyntax = nameEquals.FirstOrDefault(a =>
+                    var maxSerializedCharLengthSyntax = attributeArgNames.FirstOrDefault(a =>
                         a.NameEquals is not null
                         && a.NameEquals.Name.Identifier.ValueText == "MaxSerializedCharLength");
 
+                    var allowEmptySyntax = attributeArgNames.FirstOrDefault(a =>
+                        a.NameEquals is not null
+                        && a.NameEquals.Name.Identifier.ValueText == "AllowEmpty");
```
Comment (frankbuckley) [inline gen/DSE.Open.Values.Generators/ValueTypesGenerator.cs:203]: Should this be "AllowDefault" - empty makes sense for a value with a string or collection of values underpinning it, but not for a primitive value like int

**Review by frankbuckley — APPROVED** [review]
Can you bump version (version.json in root) to "0.7.0-alpha.{height}" for this one?

Then run ./eng/scripts/versions.ps1 before committing and merging pull request

**Review by frankbuckley — APPROVED** [review]

File: gen/DSE.Open.Values.Generators/Model/ValueTypeSpec.cs (hunk 1 above)
Comment (rabuckley) [reply gen/DSE.Open.Values.Generators/Model/ValueTypeSpec.cs]: I don't think so. Really, it should be required. It's always set to the correct state when building the spec.

File: gen/DSE.Open.Values.Generators/ValueTypesGenerator.cs (hunk 2 above)
Comment (rabuckley) [reply gen/DSE.Open.Values.Generators/ValueTypesGenerator.cs]: Done

**Pull request merged**